zerver/tests/test_invite.py

```diff
@@ -3003,6 +3003,197 @@ def test_create_multiuse_link_api_call(self) -> None:
+        ]
+
+        self.login("iago")
+        invite_expires_in_minutes = 2 * 24 * 60
+        do_create_multiuse_invite_link(
+            self.user_profile,
+            initial_invited_as,
+            invite_expires_in_minutes,
+            include_realm_default_subscriptions=False,
+            streams=initial_streams,
+        )
+        initial_multiuse_object = MultiuseInvite.objects.get()
+
+        members_system_group = NamedUserGroup.objects.get(
+            name=SystemGroups.MEMBERS, realm=realm, is_system_group=True
+        )
+        do_change_realm_permission_group_setting(
+            realm, "create_multiuse_invite_group", members_system_group, acting_user=None
+        )
+
+        # Try to patch the invite as a non-admin user with
+        # members having permission to create reusable invite
+        self.login("hamlet")
+        result = self.client_patch(
+            f"/json/invites/multiuse/{initial_multiuse_object.id}",
+            {
+                "invite_as": PreregistrationUser.INVITE_AS["MEMBER"],
+                "stream_ids": modified_streams,
+                "include_realm_default_subscriptions": orjson.dumps(False).decode(),
+            },
+        )
+        self.assert_json_success(result)
+
+        moderators_system_group = NamedUserGroup.objects.get(
+            name=SystemGroups.MODERATORS, realm=realm, is_system_group=True
+        )
+        do_change_realm_permission_group_setting(
+            realm, "create_multiuse_invite_group", moderators_system_group, acting_user=None
+        )
+
+        # Try to patch the invite as a non-admin user without
+        # members having permission to create reusable invite
+        self.login("hamlet")
+        result = self.client_patch(
+            f"/json/invites/multiuse/{initial_multiuse_object.id}",
+            {
+                "invite_as": PreregistrationUser.INVITE_AS["REALM_OWNER"],
+                "stream_ids": modified_streams,
+                "include_realm_default_subscriptions": orjson.dumps(False).decode(),
+            },
+        )
+        self.assert_json_error(result, "Insufficient permission")
+
+        self.login("iago")
+        result = self.client_patch(
+            f"/json/invites/multiuse/{initial_multiuse_object.id}",
+            {
+                "invite_as": PreregistrationUser.INVITE_AS["MODERATOR"],
+                "stream_ids": modified_streams,
+                "include_realm_default_subscriptions": orjson.dumps(False).decode(),
+            },
+        )
+        self.assert_json_success(result)
+
+        self.login("hamlet")
+        result = self.client_patch(
+            f"/json/invites/multiuse/{initial_multiuse_object.id}",
+            {
+                "invite_as": PreregistrationUser.INVITE_AS["MEMBER"],
+                "stream_ids": modified_streams,
+                "include_realm_default_subscriptions": orjson.dumps(False).decode(),
+            },
+        )
```

Comment:
This test should also check case that only user who created the user can edit the invite.